# HG changeset patch
# User Rik <rik@octave.org>
# Date 1532632105 25200
#      Thu Jul 26 12:08:25 2018 -0700
# Branch stable
# Node ID 61f8d94f6bbbecda57975170a1dbe0f0df035d87
# Parent  1ba60c0455f4cc8ae685db76982d1241e9a9f537
index.tst: Make tests case insensitive for NaN, NA, Inf values (bug #54385).

* index.tst: Use regular expression character classes to make BIST tests case
insensitive.

diff --git a/test/index.tst b/test/index.tst
--- a/test/index.tst
+++ b/test/index.tst
@@ -496,17 +496,17 @@
 %! x([], false, :) = [];
 %! assert (x, y);
 
 
   ##  Test indexing of unnamed constants
 %!error <index \(0\): subscripts must be>     1(0)
 %!error <index \(-1\): subscripts must be>    1(-1)
 %!error <index \(_,0.5\): subscripts>                 {}(1,0.5)
-%!error <index \(nan,_\): subscripts>                 1(NaN,1)
+%!error <index \([Nn][aA][Nn],_\): subscripts>                 1(NaN,1)
 %!error <index \(_,_,<cell....\[x8\]...\): subscripts> [](1,1,{},1,1,1,1,1,1,1,1)
 %!error <index \(...\[x9\]...-1,_\): subscript>      1(1,1,1,1,1,1,1,1,1,-1,1)
 %!error <index \(2\): out of bound 1>                1(2)
 %!error <index \(1\): out of bound 0>                [](1)
 %!error <index \(_,1\): but object has size 5x0>     zeros(5,0)(3,1)
 %!error <index \(3,_\): but object has size 0x5>     zeros(0,5)(3,1)
 %!error <index \(-1\): subscripts>                   1(1)(-1)(1)
 %!
@@ -542,18 +542,18 @@
 %!error <abc\(_,0.3,_\): subscripts>  abc(2,0.3,5)
 %!error <abc\(_,0.3,_\): subscripts>  abc{2,0.3,5}
 %!error <abc\(-2,_,_,_\): subscripts> abc{-2,1,1,1}
 %!error <abc\(0,_,_,_\): subscripts>  abc(0,1,1,1) = 1
 
 %! ##  Test permutation matrices
 %!shared abc
 %! abc = eye(3)([3 1 2],:);
-%!error <abc\(nan\): subscripts>         abc(NA)
-%!error <abc\(_,_,_,inf,_\): subscripts> abc(1,1,1,Inf,1)
+%!error <abc\([Nn][aA][Nn]\): subscripts>         abc(NA)
+%!error <abc\(_,_,_,[Ii][nN][Ff],_\): subscripts> abc(1,1,1,Inf,1)
 
 %! ##  Test sparse matrices
 %!shared abc
 %! abc = sparse(3,3);
 %!error <abc\(-1\): subscripts>                abc(-1)
 %!error <abc\(-1\): subscripts>                abc(-1) = 1
 %!error <abc\(-1,_\): subscripts>              abc(-1,1)
 %!error <abc\(-1,_\): subscripts>              abc(-1,1) = 1
